### DISCUSSION TITLE
Get name and username of users logged with Okta provider

### DISCUSSION THREAD

**Discussion by alexBonadei** [mia-platform/community#512] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

.

### Description

Hi,
I set up the Authentication Service with the Okta provider.
The login flow is working, but in the users crud the informations about the name and surname are empty.

How can I retrieve them?

**Comment by malta895** [comment]
Hello @alexBonadei,
Could you make sure the user collection has the same fields as [explained in the docs](https://docs.mia-platform.eu/docs/runtime_suite/authentication-service/configuration#users-collection)?
The `surname` field is not supported, and the `name` field is supposed to be filled with the complete display name of the user, i.e. name and surname.

The collection is upserted after a login, the authentication service calls the provider userinfo and populates the fields by mapping their values as follows (userinfo field => authentication server user collection):
- `sub`=> `providerUserId`
- `email`=>`email`
- `name`=>`name`
- `nickname`=>`username`

Any undefined field will fallback to the empty string, so it is likely that your Okta userinfo endpoint is not configured to return a `name` field in its payload. Could you please check that?

**Reply by alexBonadei** [reply]
Hi @malta895,

The `users` crud is correctly configured. Probably is the Okta userinfo endpoint that is not configured to return the `name` field.
The fact is that I have no access to the Okta application so I was not sure about that.

Thank you!

**Reply by davidebianchi** [reply, marked as answer]
Have you set the correct scope? I think it's the profile scope to set to retrieve the user base information.

**Reply by alexBonadei** [reply]
It worked! I had as scope just ['openid']. I added `profile` and it worked.

This part about scopes is not documented, maybe it would be nice to add some info about it!

Thank you!

**Reply by fredmaggiowski** [reply]
we can add some notes, however scope management solely depends on the provider implementation :\

**Reply by alexBonadei** [reply]
IMHO if someone is not fully aware of the authentication flow, is not clear the use of the scopes.

In the Okta configuration example there is this: 
```
"scope": ["your_scope"],
```
I thought that could be a custom value, but as you said that value depends on the Okta allowed ones. Probably just a link to this page would help a lot. https://developer.okta.com/docs/api/oauth2/

**Reply by fredmaggiowski** [reply]
yeah sure, we can even update the example to be more insightful, do you want to try and submit a PR to update the contents with your experience?
